website/build.gradle

```diff
@@ -170,19 +188,14 @@ task publishWebsite << {
     println "Adding website-publish remote"
     // Cannot authenticate to the default github uri, so specify gitbox.
     git.remote.add(name: 'website-publish',
-                   url: 'https://gitbox.apache.org/repos/asf/beam.git',
-                   pushRefSpecs: ['refs/heads/asf-site'])
+                   url: 'https://gitbox.apache.org/repos/asf/beam.git')
   }
+
   // Because git.push() fails to authenticate, run git push directly.
-  exec {
-    executable 'sh'
-    args '-c', "git push website-publish asf-site"
-  } 
-  // Remove the remote. grgit does not have a remote.remove method.
-  exec {
-    executable 'sh'
-    args '-c', "git remote remove website-publish"
-  }
+  shell "git fetch website-publish asf-site"
+  shell "git merge website-publish/asf-site --no-edit"
+  shell "git push website-publish asf-site < /dev/null"
```

Comment:
I just checked the documentation for `git push` and it can take a URL for its remote parameter, meaning we can forego all of this remote config management. Then, the command becomes: `git push https://gitbox.apache.org/repos/asf/beam.git asf-site` and the rest can be removed.